vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc

```diff
@@ -1644,4 +1649,161 @@ void LedgerImpl::FetchBalance(ledger::FetchBalanceCallback callback) {
   bat_wallet_->FetchBalance(callback);
 }
 
+void LedgerImpl::OnGetAllTransactions(
+    ledger::mojom::AllTransactionsPtr transaction_list,
+    uint32_t record,
+    ledger::ACTIVITY_MONTH month,
+    uint32_t year,
+    double current_balance,
+    ledger::MonthlyStatementCallback callback) {
+  for (uint32_t ix = 0; ix < transaction_list->publishers.size(); ix++) {
+    transaction_list->publishers[ix]->verified =
+        bat_publishers_->isVerified(transaction_list->publishers[ix]->id);
+  }
+
+  ledger::BalanceReportPtr report = ledger::BalanceReport::New();
+  ledger::BalanceReportInfo report_info;
+  if (bat_publishers_->getBalanceReport(month, year, &report_info)) {
+    report->grants = report_info.grants_;
+    report->earning_from_ads = report_info.earning_from_ads_;
+    report->auto_contribute = report_info.auto_contribute_;
+    report->recurring_donation = report_info.recurring_donation_;
+    report->one_time_donation = report_info.one_time_donation_;
+    report->opening_balance = report_info.opening_balance_;
+    report->closing_balance = report_info.closing_balance_;
+    report->deposits = report_info.deposits_;
+    report->total = report_info.total_;
+    bat_publishers_->setBalanceReport(
+        month,
+        year,
+        report_info);
+    if (report->grants != "0") {
+      ledger::mojom::TransactionStatementInfoPtr transaction =
+          ledger::mojom::TransactionStatementInfo::New();
+      transaction->probi = report->grants;
+      transaction->category = 3;  // placeholder for grants on statement
+      base::Time parsed_time;
+      std::string time_str =
+          base::StringPrintf("%s/1/%s",
+            std::to_string(month).c_str(),
```

Comment:
Probably simpler to use the `%u` format specifier instead of `%s`, then you don't need to convert to string.

Reply:
👍 done